src/main/java/com/connexta/search/index/IndexManagerImpl.java

```diff
@@ -47,9 +53,10 @@ public void index(
 
     final Index index;
     try {
-      index = new Index(productId, IOUtils.toString(inputStream, StandardCharsets.UTF_8));
+      JsonNode jsonNode = parseJson(file.getInputStream());
+      index = new Index(productId, getElement(jsonNode, EXT_EXTRACTED_TEXT));
```

Comment:
✏️ Move this Index instantiation out of this try catch. If the Index constructor throws a IOException, we wouldn't want to throw the IndexException in the catch block.